homeassistant/components/velbus/__init__.py

```diff
@@ -121,25 +99,23 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
 class VelbusEntity(Entity):
     """Representation of a Velbus entity."""
 
-    def __init__(self, module, channel):
+    def __init__(self, channel):
         """Initialize a Velbus entity."""
-        self._module = module
         self._channel = channel
 
     @property
     def unique_id(self):
         """Get unique ID."""
-        serial = 0
-        if self._module.serial == 0:
-            serial = self._module.get_module_address()
-        else:
-            serial = self._module.serial
-        return f"{serial}-{self._channel}"
+        return "{}.{}.{}".format(
+            self._channel.get_module_type(),
+            self._channel.get_module_address(),
+            self._channel.get_channel_number(),
+        )
```

Comment:
This is indeed something I should fix, making sure the unique Id stays the same after an upgrade. This will make the upgrade almost invisible for the end user